packages/graph/hash_graph/lib/graph/src/store/query/mod.rs

```diff
@@ -147,106 +167,177 @@ impl Expression {
+                            literal => bail!(
+                                Report::new(ExpressionError)
+                                    .attach_printable(format!("Not a boolean: {literal:?}"))
+                            ),
                         }
                     }
                     Literal::Bool(true)
                 }
                 Expression::Any(expressions) => {
                     for expression in expressions {
-                        match expression.evaluate(resolver, context).await {
+                        match expression.evaluate(resolver, context).await? {
                             Literal::Bool(false) => continue,
-                            literal @ Literal::Bool(true) => return literal,
-                            literal => panic!("Not a boolean: {literal:?}"),
+                            literal @ Literal::Bool(true) => return Ok(literal),
+                            literal => bail!(
+                                Report::new(ExpressionError)
+                                    .attach_printable(format!("Not a boolean: {literal:?}"))
+                            ),
                         }
                     }
                     Literal::Bool(false)
                 }
                 Expression::Literal(literal) => literal.clone(),
-                Expression::Path(path) => resolver.resolve(&path.segments, context).await,
-                Expression::Field(_identifier) => {
-                    // see https://app.asana.com/0/0/1202884883200943/f
-                    todo!("`Literal::Object`")
-                }
-            }
+                Expression::Path(path) => resolver
+                    .resolve(&path.segments, context)
+                    .await
+                    .change_context(ExpressionError)?,
+                Expression::Field(_) => bail!(
+                    Report::new(ResolveError::UNIMPLEMENTED_LITERAL_OBJECT)
+                        .change_context(ExpressionError)
+                ),
+            })
         }
         .boxed()
     }
 }
 
-/// Resolves this types into a [`Literal`].
-// TODO: DOC
-//   see https://app.asana.com/0/0/1202884883200976/f
+#[derive(Debug)]
+pub enum ResolveError {
+    Unimplemented {
+        description: &'static str,
+        issue: &'static str,
+    },
+    EmptyPath {
+        literal: Literal,
+    },
+    CannotIndex {
+        path: Path,
+        literal: Literal,
+    },
+    StoreReadError,
+}
+
+impl fmt::Display for ResolveError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Unimplemented { description, issue } => write!(fmt, "{description}, see {issue}"),
+            Self::EmptyPath { literal } => write!(fmt, "empty path when resolving `{literal:?}`"),
+            Self::CannotIndex { path, literal } => write!(
+                fmt,
+                "cannot index `{literal:?}` with path `{}`",
+                path.segments
```

Comment:
Can we turn this into a method on `Path`? 

I don't like the idea of us rebuilding it in some other module. If we end up changing the syntax or path separators and whatnot, I could easily see us missing it here.

Reply:
I added a `Display` implementation: cea45f477c328a515c78c6cc228bc13f570afae7